nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearch5.java

```diff
@@ -129,6 +130,12 @@
             .defaultValue("100")
             .expressionLanguageSupported(true)
             .build();
+    
+    public static final PropertyDescriptor VERSION_ATTRIBUTE = new PropertyDescriptor.Builder()
+			.name("Version").displayName("Version Attribute")
+			.description("The name of the attribute containing the version for each FlowFile. Only used for index operation. Forces External Versioning")
+			.required(false).expressionLanguageSupported(false).addValidator(StandardValidators.ATTRIBUTE_KEY_VALIDATOR)
```

Comment:
Idea here was to follow the same logic as the ID_ATTRIBUTE so not to break pattern. But not having the field required, so you can Update/Upsert documents.
Actually I think the Id should be expression enabled and even allowed to be null - You can index docs without an ID.

What change would you sugest ?

Reply:
That was an unfortunate oversight on my part, thinking you'd need a value for ID, and we've been a bit stuck with it ever since (until the next major version release perhaps). If the value is not required, then I think EL support should suffice?

Reply:
What do you think about the above suggestion, changing from an attribute name to an Expression Language supported property with a NON_EMPTY_VALIDATOR? Personally I'd prefer it that way, even if it doesn't behave the same as ID_ATTRIBUTE, since they are not really used the same way anyway. A regular EL-supporting property is more flexible and you can still use a flow file attribute by simply wrapping the name in an expression like ${myAttribute}.
If you'd like to keep the current behavior and/or discuss further, please let me know

Reply:
I think that's the way to go !

It would offer the EL capabilities support, will be handy.
(Let me just recap) We'd enable Expression Languange support, validated by an NON_EMPTY_VALIDATOR. Effectivly allowing for:
- No Versioning (if Version is not specified)
- Versioning (if specified)
- New IDs (if ID is not specified)
- Using IDs (if specified)
(Assuming we apply across both ID and Version)

What do you think ?

I'll have a go at this and push some code once I have a bit of free time :)

Reply:
Well, the Identifier Attribute property is the way people have been using IDs, so making this change now would be a bit too invasive IMO. I think since the Versioning is new we can do it for that, but please keep the Identifier Attribute the same and we will make the appropriate change for NiFi 2.0 (please feel free to file a Jira). I apologize for my short-sightedness on the latter, but I don't think we can absorb that kind of breaking change on a minor release.